### PR TITLE
[Utility] feat: add utilities to notify slack about a new library release.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

This is how such an automatic notification would look like (internal link): https://huggingface.slack.com/archives/C03HBN1C8CW/p1707889851090899.

image

Steps to create the webhook-enabled Slack Bot (documenting in case someone else finds it useful):

* Go to the Slack API website and create a new app.
* Choose the workspace you want to install your app to.
* In the 'Add features and functionality' section, choose 'Incoming Webhooks'.
* Activate incoming webhooks and create a new webhook for the channel you want to post messages to.
* Copy the webhook URL; you'll use it in your Python script to send messages to Slack.

### CODE REVIEW

**Comment by sayakpaul** [comment]
This is wrongly crafted. In its current form, the bot will always notify the channel within a fixed interval. We need to make it conditional on the PyPI release status instead. Will look into it.

**Comment by LysandreJik** [comment]
cc @ydshieh who might have tips/comments about the Slack-related work (he's handling it from the transformers' POV)

**Comment by sayakpaul** [comment]
@ydshieh a gentle ping in case you want to exchange notes :)

File: scripts/notify_slack.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return None
+
+
+def get_github_release_info(github_repo):
+    """Fetch the latest release info from GitHub."""
+    url = f"https://api.github.com/repos/{github_repo}/releases/latest"
+    response = requests.get(url)
+    if response.status_code == 200:
+        data = response.json()
+        return {"tag_name": data["tag_name"], "url": data["html_url"]}
```
Comment (yiyixuxu) [inline scripts/notify_slack.py:29]:
```suggestion
        return {"tag_name": data["tag_name"], "url": data["html_url"], "release_time" : data["published_at"}
```

we can maybe:
1. extract the published time the latest release here
2. compare with the current time and notify if it is less than 12 hours?

**Comment by ydshieh** [comment]
Hi @sayakpaul 

I love the idea of having such notification!

In `transformers`, it uses `from slack_sdk import WebClient` and `client.chat_postMessage`, (it was that way before I joined), which is different from what you have done in this PR. So I won't have comments to provide here regarding this part.

As you know and mentioned by @yiyixuxu , the condition has to be adjusted. Also, at the end, we need to decide which libraries to watch for the notifications.

I would rather fail the script if the information could not be fetched instead of just printing a message.

**Comment by sayakpaul** [comment]
Closing this with https://github.com/huggingface/diffusers/pull/7270.

**Pull request closed without merging**